lib/song.rb

```diff
@@ -0,0 +1,60 @@
+class Song
+  extend Concerns::Findable
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    @artist = artist
+    if artist
+      artist.add_song(self) unless artist.songs.include?(self)
+    end
+    @genre = genre
+    if genre
+      genre.songs << self unless genre.songs.include?(self)
+    end
+  end
+
+  attr_accessor :name
```

Comment:
Consider putting the *attr_accessor, attr_reader* etc. at the beginning of the class.